perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_resource_attribute_def_def_defaultDataQuotas.java

```diff
@@ -22,16 +30,52 @@
  */
 public class urn_perun_resource_attribute_def_def_defaultDataQuotas extends ResourceAttributesModuleAbstract implements ResourceAttributesModuleImplApi {
 
+	public static final String A_R_maxUserDataQuotas = AttributesManager.NS_RESOURCE_ATTR_DEF + ":maxUserDataQuotas";
+
 	@Override
 	public void checkAttributeValue(PerunSessionImpl perunSession, Resource resource, Attribute attribute) throws InternalErrorException, WrongAttributeValueException, WrongReferenceAttributeValueException, WrongAttributeAssignmentException {
 		//attribute can be null, it means there are no default settings on resource
-		if(attribute.getValue() == null) {
+		if (attribute.getValue() == null) {
 			return;
 		}
 
 		//Check if every part of this map has the right pattern
 		//And also check if every quota part has right settings (softQuota<=hardQuota)
-		perunSession.getPerunBl().getModulesUtilsBl().checkAndTransferQuotas(attribute, resource, null, true);
+		Map<String, Pair<BigDecimal, BigDecimal>> defaultDataQuotasForResource = perunSession.getPerunBl().getModulesUtilsBl().checkAndTransferQuotas(attribute, resource, null, true);
+
+		//If there are no values after converting quota, we can skip testing against maxUserDataQuota attribute, because there is nothing to check
+		if (defaultDataQuotasForResource == null || defaultDataQuotasForResource.isEmpty()) return;
+
+		//Get maxUserDataQuotas value on this resource
+		Attribute maxUserDataQuotasAttribute;
+		try {
+			maxUserDataQuotasAttribute = perunSession.getPerunBl().getAttributesManagerBl().getAttribute(perunSession, resource, A_R_maxUserDataQuotas);
+		} catch (AttributeNotExistsException ex) {
+			throw new ConsistencyErrorException(ex);
+		}
+
+		//Check and transfer maxUserDataQuotasForResource
+		Map<String, Pair<BigDecimal, BigDecimal>> maxUserDataQuotasForResource;
+		try {
+			maxUserDataQuotasForResource = perunSession.getPerunBl().getModulesUtilsBl().checkAndTransferQuotas(maxUserDataQuotasAttribute, resource, null, true);
+		} catch (WrongAttributeValueException | InternalErrorException ex) {
```

Comment:
InternalErrorException means that maxUserQuota is not in correct format?

Reply:
Yes, because the only way how InternalError can be thrown there is by problem with "null" value in quotas attribute.

Reply:
And of course if place holder is null.

Reply:
The problem with null value in quotas attributes should be handled in that attribute check module.

Not sure, what you mean by "place holder is null".

Reply:
Ok, I will let throw InternalErrorException where it is expected in modules.